homeworks/17/chat-system/src/main/java/org/example/ClientHandler.java

```diff
@@ -0,0 +1,153 @@
+    }
+
+    private void setBufferedWriter(OutputStream outputStream) {
+        this.bufferedWriter = new BufferedWriter(new OutputStreamWriter(outputStream));
+    }
+
+    private void setSocket(Socket socket) {
+        if (Objects.isNull(socket)) {
+            throw new InvalidParameterException("Client socket cant be null");
+        }
+        this.socket = socket;
+    }
+
+    private void closeEverything(Socket socket, BufferedReader bufferedReader, BufferedWriter bufferedWriter) {
+        try {
+            closeWriter(bufferedWriter);
+        } catch (IOException exception) {
+            exception.printStackTrace();
+        }
+        try {
+            closeReader(bufferedReader);
+
+        } catch (IOException exception) {
+            exception.printStackTrace();
+        }
+        try {
+            closeSocket(socket);
+
+        } catch (IOException exception) {
+            exception.printStackTrace();
+        }
+        removeHandler();
+    }
+
+    private void closeReader(BufferedReader bufferedReader) throws IOException {
+        if (Objects.nonNull(bufferedReader)) {
+            bufferedReader.close();
+        }
+    }
+
+    private void closeWriter(BufferedWriter bufferedWriter) throws IOException {
+        if (Objects.nonNull(bufferedWriter)) {
+            bufferedWriter.close();
+        }
+    }
+
+    private void closeSocket(Socket socket) throws IOException {
+        if (Objects.nonNull(socket)) {
+            socket.close();
+        }
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public void broadcastMessage(String messageToSend) {
+        if (Objects.isNull(messageToSend) || messageToSend.startsWith("/quit")) {
+            closeEverything(this.socket, bufferedReader, bufferedWriter);
+            broadcastAnnouncement("SERVER: " + username + " has ritna kambanata, rip bratan o7!");
+        } else if (messageToSend.startsWith("/rename ")) {
+            broadcastAnnouncement("SERVER: " + username + " has changed their name to " + messageToSend.substring("/rename ".length()));
+            username = messageToSend.substring("/rename ".length());
+        } else if (messageToSend.startsWith("/message ")) {
+
+            for (ClientHandler clientHandler : clientHandlers) {
+                try {
+                    if (!clientHandler.username.equals(username)) {
+                        clientHandler.bufferedWriter.write("[" + DateTimeFormatter.ofPattern("HH:mm a").format(LocalTime.now()) + "] ");
+                        clientHandler.bufferedWriter.write(username + ": " + messageToSend.substring("/message ".length()));
+                        clientHandler.bufferedWriter.newLine();
+                        clientHandler.bufferedWriter.flush();
+                    }
+                } catch (IOException exception) {
+                    closeEverything(this.socket, bufferedReader, bufferedWriter);
+                }
+            }
+        }
+    }
```

Comment:
I think it is a good idea if the command is wrong to tell the user which command they can use.